apps/etterlatte-brev-api/src/main/kotlin/no/nav/etterlatte/brev/VedtaksbrevService.kt

```diff
@@ -124,16 +197,50 @@ class VedtaksbrevService(
         return db.slett(id)
     }
 
-    private suspend fun genererPdf(behandling: Behandling, mottaker: Mottaker): ByteArray {
-        val (avsender, attestant) = adresseService.hentAvsenderOgAttestant(behandling.vedtak)
-
-        val brevRequest = BrevbakerRequest.fra(behandling, avsender, mottaker, attestant)
-
+    private suspend fun genererPdf(brevID: BrevID, brevRequest: BrevbakerRequest): ByteArray {
         val brevbakerResponse = brevbaker.genererPdf(brevRequest)
         val brev = Base64.getDecoder().decode(brevbakerResponse.base64pdf)
 
-        logger.info("Generert brev for vedtak (vedtakId=${behandling.vedtak.id}) med størrelse: ${brev.size}")
+        logger.info("Generert brev (id=$brevID) med størrelse: ${brev.size}")
 
         return brev
     }
+
+    private fun getJsonFile(url: String) = javaClass.getResource(url)!!.readText()
+
+    private fun hentInitiellPayload(sakType: SakType, vedtakType: VedtakType): Slate {
+        return when (sakType) {
+            SakType.OMSTILLINGSSTOENAD -> {
+                when (vedtakType) {
+                    VedtakType.INNVILGELSE -> getJsonFile("/maler/oms-nasjonal-innvilget.json")
+                    VedtakType.OPPHOER,
+                    VedtakType.AVSLAG,
+                    VedtakType.ENDRING -> getJsonFile("/maler/tom-brevmal.json")
+                }
+            }
+
+            SakType.BARNEPENSJON -> {
+                when (vedtakType) {
+                    VedtakType.INNVILGELSE,
+                    VedtakType.OPPHOER,
+                    VedtakType.AVSLAG,
+                    VedtakType.ENDRING -> getJsonFile("/maler/tom-brevmal.json")
+                }
+            }
+        }.let { deserialize(it) }
+    }
+
+    private fun hentProsessType(sakType: SakType, vedtakType: VedtakType): BrevProsessType {
+        return when (sakType) {
+            SakType.OMSTILLINGSSTOENAD -> BrevProsessType.MANUELL
```

Comment:
Basert på koden ovenfor, bør det ikke være automatisk for noen av vedtakstypene for OMS også?

Reply:
Henting av payload? Det er bare en fallback for å ikke krasje. Så at vi under testing i dev kan lage dummybrev.